zkevm-circuits/src/tx_circuit.rs

```diff
@@ -297,14 +313,60 @@ impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize>
                 }
                 Ok(())
             },
-        )?;
+        )
+    }
+}
+
+impl<F: Field> SubCircuit<F> for TxCircuit<F> {
+    type Config = TxCircuitConfig<F>;
+
+    fn new_from_block(block: &witness::Block<F>) -> Self {
+        let mut rng = ChaCha20Rng::seed_from_u64(42);
```

Comment:
I think it would be nice to either change the `new_from_block` constructor to also take a Rng or put that into the witness struct.

Reply:
This rng is used to create an auxiliary point that is required for the ECDSA Chip.  This point is assigned as witness, and is not seen by the verifier.  I checked with @davidnevadoc and it's OK to reuse the same point through different instantiations of the circuit; that's why I moved the rng to this function instead of taking it externally.  But I understand that this looks weird, like a hardcoded randomness that should only be acceptable in test code.  I'll check if I can replace the generation of a random point with an rng, by adding a specific point directly.

Reply:
Actually it's not clear if the EcdsaChip is safe with this random point passed as witness.  David started investigating this https://github.com/privacy-scaling-explorations/halo2wrong/issues/53 
But in any case, either a hardcoded point can be used, or the EcdsaChip needs to be updated, but the `new_from_block` shouldn't be concerned with any randomness for now :)